packages/bpk-component-skeleton/src/BpkSkeleton.tsx

```diff
@@ -0,0 +1,52 @@
+/*
+ * Backpack - Skyscanner's Design System
+ *
+ * Copyright 2016 Skyscanner Ltd
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { cssModules } from '../../bpk-react-utils';
+
+import BpkBaseSkeleton from './BpkBaseSkeleton';
+import { SKELETON_TYPES, IMAGE_SKELETON_STYLE, SIZE_TYPES } from './common-types';
+
+import type { ComponentProps} from './common-types';
+
+import STYLES from './BpkSkeleton.module.scss';
+
+
+const getClassName = cssModules(STYLES);
+
+const BpkSkeleton = (props: ComponentProps) => {
+  const { size, type } = props;
+
+  const isImageRounded = type === SKELETON_TYPES.image && props.style === IMAGE_SKELETON_STYLE.rounded;
+  const classNames: string = getClassName(
+      `bpk-skeleton__${type}`,
+      typeof size !== 'object' && `bpk-skeleton__${type}--${size}`,
+      isImageRounded && 'bpk-skeleton__image--rounded'
+   );
+  const styleObj = typeof size === 'object' ? size : undefined;
+
+  return (
+    <BpkBaseSkeleton skeletonStyle={classNames} styleObj={styleObj} />
+  )
+};
+
+BpkSkeleton.defaultProps = {
+  size: SIZE_TYPES.default,
+  style: IMAGE_SKELETON_STYLE.default,
+}
```

Comment:
In TS we wouldn't declare default props like this but more up in the props definition on L31

Reply:
The reason why I don't deconstruct at line L31 is that it will cause ts error. (The style property only exist when type is 'image')
<img width="956" alt="image" src="https://github.com/Skyscanner/backpack/assets/101770212/9d268d27-8c55-41cc-87ce-9ed9110a3c2c">

Reply:
Here what we can do is what you had before before

`const BpkSkeleton = ({type, size = SiZE_TYPES.default, style = IMAGE_SKELETON_STYLE.default})` 

Then you can remove the const {} = props declaration and just use the props as normal :) 

Then